File: 007/Haskell/Problem007Test.hs
module Main where

import Test.HUnit
import System
import Problem007

nthPrimeTests = TestList [
  TestCase (2 @=? nthPrime 1),
  TestCase (3 @=? nthPrime 2),
  TestCase (5 @=? nthPrime 3),
  TestCase (7 @=? nthPrime 4),
  TestCase (11 @=? nthPrime 5),
  TestCase (13 @=? nthPrime 6),
  TestCase (1 @=? 1)
  ]
  
tests = TestList [
	TestLabel "testing nthPrime" nthPrimeTests,
	TestCase (1 @=? 1)
	]
	
main = runTestTT tests >>= checkTests
  where
    checkTests counts = case counts of
    	Counts _ _ errors failures ->
    		if (errors + failures > 0)
    		then exitFailure
    		else (exitWith ExitSuccess)

File: 007/Ruby/problem007_test.rb
require "test/unit"

require_relative "problem007"

class TestProblem007 < Test::Unit::TestCase

  def test_nth_prime
    assert_equal 2, 1.nth_prime
    assert_equal 3, 2.nth_prime
    assert_equal 5, 3.nth_prime
    assert_equal 7, 4.nth_prime
    assert_equal 11, 5.nth_prime
    assert_equal 13, 6.nth_prime
  end

end
